edgelet/hsm-sys/azure-iot-hsm-c/CMakeLists.txt

```diff
@@ -71,6 +72,12 @@ if(MSVC)
     add_definitions(-DGB_MEASURE_MEMORY_FOR_THIS -DGB_DEBUG_ALLOC)
 else()
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -fPIC -Wall -Werror -Wextra -Wformat=2 -Wformat-security -std=c99 ")
+
+    # suppress compiler warnings on macos complaining about calls to vsnprintf
+    if(APPLE)
```

Comment:
Would it make sense to condition this on compiling *with clang* (via `CMAKE_COMPILER_ID`) instead of *on macos* ? In case someone is building it with clang on Linux and needs the same suppression.

Reply:
Agree.